### PR TITLE
Changes after testing against DevStack

### PR DESCRIPTION
**Pull request by ClifHouck**
Apologies again for a somewhat unfocused commit.

* Added a DevStack scout which should work for testing Aresnal in
        DevStack using the example agent driver Ironic DevStack configuration.
* A few small tweaks to OnMetalScout which solves issues found in
         testing.
* Small change to the DirectorScheduler to tie updating Ironic node
        state to generating strategy directives.
* Made the SimpleProportionalStrategy percentage a configuration option
        instead of a class argument.
* Change SimpleProportionalStrategy behavior to not try to
        cache 'fractional' nodes in order to meet or exceed the
        percentage directive. This avoids behavior where Arsenal will
        always cache the last unprovisioned node in order to meet its
        caching goal, in exchange for not always strictly meeting the
        caching goal.
* Updated SimpleProportionalStrategy tests to match behavior.
* Added an example arsenal configuration file.

### CODE REVIEW

File: arsenal/director/onmetal_scout.py [hunk 1]
```diff
@@ -163,23 +163,26 @@ def _find_glance_image(self, cache_node_action):
...
 
-        args = {
+        args = ({
             'image_info': {
                 'id': cache_node_action.image_uuid,
                 'urls': [CONF.glance.api_endpoint + image_url],
                 'checksum': cache_node_action.image_checksum
             }
-        }
+        })
```
Comment (jimrollenhagen) [inline arsenal/director/onmetal_scout.py]: Why the extra brackets? They aren't doing anything... If you intend for this to be a tuple it needs a comma in there, e.g. `({},)`

Comment (jimrollenhagen) [inline arsenal/director/onmetal_scout.py]: Also, should this be passed into call() as `*args`?

Comment (jimrollenhagen) [inline arsenal/director/onmetal_scout.py]: Lastly... I guess if this is working you can ignore me :)

**Comment by jimrollenhagen** [comment]
Assuming the bit of code I commented on makes sense (which isn't clear to me), +1

**Comment by jimrollenhagen** [comment]
Still +1 on this :D

**Comment by pcsforeducation** [comment]
+1 as well, nice!

**Pull request merged**